### ISSUE TITLE
New Python interpreter selection combobox displays previous, not actual executable path if selected from file browser

### ISSUE THREAD

**Issue by CAM-Gerlach** [spyder-ide/spyder#7674]
## Problem Description

As a follow-on to the work of @dalthviz and @ccordoba12 in  #7566 , I just tested this on the latest ``3.x`` on Windows 8.1, and there appears to be a UI bug with this: when selecting a new interpreter from the file browser (i.e. rather than choosing it from the dropdown list), and then hitting "Apply" and "OK", just "OK", "Apply" then closing the dialog, etc., the new setting does work (i.e. new consoles are opened in the environment the user selected). However, if I re-open preferences, the interpreter path displayed in the combobox is that of the interpreter I previously had selected before the aforementioned change, rather than the actual one. However, so long as one does not actually interact with this field, making other changes in the preferences and then saving makes the interpreter for new consoles stay the one the user selected, rather than the one displayed, and selecting interperters from the list *does* make them display correctly in the box once preferences is re-opened.

Also, conversely, when selecting one or multiple new interpreters from the file browser dialog that aren't currently in the list, their path(s) is (are) not added to the list, even if the Apply button is clicked to set them, until the preferences dialog is closed and re-opened.

### What steps reproduce the problem?

1. Open ``Preferences > Python Interpreter > Use the following interpreter``
2. Select a new interpreter by navigating to the ``python`` executable in the file browser dialog; observe the path *does* update in the combobox
3. Press "OK", "Apply" then "OK", "Apply" then "X", etc.
4. Open a new IPython console. It opens in the environment selected
5. Re-open preferences and observe that the path shown is to the previous interpreter
6. Make a change in another pane and press Apply/OK/etc.
7. Open a new IPython console; the environment is still correct
8. Re-open preferences and select a different interpreter, this time from the drop-down menu
9. Save and close and open a new Console; it also opens in the correct environment
10. Re-open preferences, the correct interpreter is shown

Also, for the possibly related issue:
1. Open ``Preferences > Python Interpreter > Use the following interpreter``
2. Select a new interpreter *not* previously listed by navigating to the ``python`` executable in the file browser dialog; observe it is displayed but not included in the dropdown list
3. Click "Apply"; it is still not listed
3. Select another new, unlisted interpreter; observe neither it nor the previous is listed
5. Click "Apply", the same is still true
6. Press "OK", "Apply" then "OK", "Apply" then "X", etc.
7. Re-open preferences and observe that all previously selected interpreters are now listed; the list has now updated

### What is the expected output? What do you see instead?

Expected: Combobox to always display the selected, actual interpreter, and (ideally) the list to update to include newly selected executables at least after clicking "Apply".

Actual: If preferences is closed and re-opened, the previous executable path is displayed; *unless* console is closed and re-opened, newely added paths are *not* included in the list even if "Apply" is clicked.

## Versions

* Spyder version: 3.3.2-dev0 (``3b2cbcb2``)
* Python version: 3.6.6
* Qt version: 5.9.4
* PyQt version: 5.9.2
* Operating System name/version: Windows 8.1 x64

**Comment by ccordoba12** [comment]
@dalthviz, please work on this one.

**Comment by CAM-Gerlach** [comment]
Also, some additional related issues and notes:

* The above persists even after a restart; i.e. the interpreter actually launched is correct, but the path displayed in preferences is wrong.
* Using the file browser or manually entering the path to the same Python interpreter Spyder is currently running under will appear to work just like steps 1-3 above (i.e., it will display in the combobox, and clicking "Apply" and "OK" will appear to work and give no errors), but it will in fact silently fail and will not actually change the interpreter path displayed or actually used, which is terrible UX and leads to the potential for considerable user frustration if the user happens to pick the Python interpreter Spyder is running. I'd actually thought I'd discovered a much more severe bug since this was the exact same behavior as the above, except the interpreter didn't actually switch either, and so did a bunch of testing and had written up a whole comment until I figured out what was likely going on. This should be handled by either allowing Spyder's own interpreter to be set (since different Spyder instances may be started in different environments, but the user may want them all to start with the same consistent interpreter for example), or at the very least giving an error popup rather than just failing silently and leaving the user to figure out  what the heck is happening.
* When I restarted and tried to select the same Python 2 interpreter executable that was listed and worked just fine before (though I got the warning message dialog when selecting it, I was able to launch consoles and run commands without any problem), it suddenely wouldn't let me select it by any means and kept giving me the "This is a not a valid Python interpreter" message and resetting it back to the default, no matter what I did. However, when I restarted again and tried it by various means, it again gave me the warning but allowed me to select it, and again it worked fine. However, I'm not yet able to determine what consistenly causes the seemingly arbitrary difference in behavior between sessions launched consecutively in the same environment.
